plugin/Form/FormAddressCreator.php

```diff
@@ -236,4 +242,128 @@ private function getUserNameById(string $userId): string
 			return '';
 		}
 	}
+
+	/**
+	 * @param FormData $pFormData
+	 * @param int $addressId
+	 * @return string
+	 * @throws ApiClientException
+	 * @throws UnknownFieldException
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 */
+	private function handleLogicMessageDuplicateAddressData(FormData $pFormData, int $addressId): string
+	{
+		$requestParams = $this->getAddressDataForApiCall($pFormData);
+		if (isset($requestParams['gdprcheckbox']) && $requestParams['gdprcheckbox']){
+			$requestParams['DSGVOStatus'] = "speicherungzugestimmt";
+		}
+		unset($requestParams['gdprcheckbox']);
+		if (key_exists('newsletter', $requestParams)) {
+			unset($requestParams['newsletter']);
+		}
+
+		$oldAddressData = $this->getDataAddressById($addressId, $requestParams);
+		unset($oldAddressData['id']);
+
+		return $this->generateMessageDuplicateAddressData($oldAddressData);
+	}
+
+	/**
+	 * @param FormData $pFormData
+	 * @param int $addressId
+	 * @param mixed $latestAddressIdOnEnterPrise
+	 * @return string
+	 * @throws ApiClientException
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 * @throws UnknownFieldException
+	 */
+	public function getMessageDuplicateAddressData(FormData $pFormData, int $addressId, $latestAddressIdOnEnterPrise): string
+	{
+		if ($addressId > $latestAddressIdOnEnterPrise || is_null($latestAddressIdOnEnterPrise)) {
+			return '';
+		}
+
+		return $this->handleLogicMessageDuplicateAddressData($pFormData, $addressId);
+	}
+
+
+	/**
+	 * @return mixed
+	 * @throws ApiClientException
+	 * @throws DependencyException
+	 * @throws NotFoundException
+	 */
+	public function getLatestAddressIdInOnOfficeEnterprise()
+	{
+		$pContainerBuilder = new ContainerBuilder;
+		$pContainerBuilder->addDefinitions(ONOFFICE_DI_CONFIG_PATH);
+		$pContainer = $pContainerBuilder->build();
+		$pContainer->get(CacheHandler::class)->clear();
+
+		$requestParams = [
+			'sortby' => 'KdNr',
+			'sortorder' => 'DESC',
+			'listlimit' => 1
+		];
+		$pApiClientAction = new APIClientActionGeneric
+			($this->_pSDKWrapper, onOfficeSDK::ACTION_ID_READ, 'address');
+		$pApiClientAction->setParameters($requestParams);
+		$pApiClientAction->addRequestToQueue()->sendRequests();
+		$result = $pApiClientAction->getResultRecords();
+
+		if (empty($result)) {
+			return null;
+		}
+
+		return $result[0]['id'];
+	}
+
+	/**
+	 * @param int $addressId
+	 * @param array $params
+	 * @return array
+	 * @throws ApiClientException
+	 */
+	private function getDataAddressById(int $addressId, array $params): array
+	{
+		$requestParams = [
+			'data' => array_keys($params)
+		];
+
+		$pApiClientAction = new APIClientActionGeneric
+			($this->_pSDKWrapper, onOfficeSDK::ACTION_ID_READ, 'address');
+		$pApiClientAction->setParameters($requestParams);
+		$pApiClientAction->setResourceId((string) $addressId);
+		$pApiClientAction->addRequestToQueue()->sendRequests();
+		$result = $pApiClientAction->getResultRecords();
+
+		return $result[0]['elements'];
```

Comment:
Is it possible to get a readable name here for email, phone and so on? If I test this I get this Data:
Vorname: anja
Name: test
email__1009: duplicate@gmail.com

instead of 
E-Mail: duplicate@gmail.com

Reply:
I fixed it. It can show a readable name here for email.
https://files.fm/u/vn72739yjv#/view/8qfu99ews6